bytestring/js/src/ByteStringsJs.kt

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2017-2024 JetBrains s.r.o. and respective authors and developers.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENCE file.
+ */
+
+package kotlinx.io.bytestring
+
+import org.khronos.webgl.ArrayBuffer
```

Comment:
This is something that should probably be addressed for all kotlinx libraries: should everyone use these built-in types, which are pretty weird considering we generally don't want anything to do with WebGL, or should they come from kotlin-wrappers, which tend to be the most used?

The question is important because the only "interop" between same types coming from different libs is unsafe-casting.

Reply:
If an `ArrayBuffer` originating from `kotlin-wrappers` needs to be converted to/from `ByteString`, then it should be done either in `kotlin-wrappers`, or in a separate library.

Reply:
It's reasonable. However I'm thinking that since most people on K/JS are using kotlin-wrappers (and most commonly its `kotlin-js` module), it would be a surprise to them that the code is red in the IDE when calling `Int8Array.toByteString()`, for example.

It's a bit of an unfortunate situation because it's all fragmented into multiple places.

1. The `org.khronos.webgl` package is explicitly [documented](https://github.com/JetBrains/kotlin/blob/5f6fd3b748204bd5b054310acb56cafd3de6fbf4/libraries/stdlib/src/Module.md?plain=1#L162) as a wrapper for WebGL, so the question "why am I using WebGL types?" would come up fairly naturally.

2. The `org.kronos.webgl` package is a remnant of the extraction ([KT-35973](https://youtrack.jetbrains.com/issue/KT-35973)) of DOM APIs from the stdlib, and there is also an open issue about moving away from it ([KT-57303](https://youtrack.jetbrains.com/issue/KT-57303)). So maybe using kotlin-wrapper types ensures better stability on the long run.